unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/DecentralandEntity.cs

```diff
@@ -180,14 +181,27 @@ public void RemoveSharedComponent(System.Type targetType, bool triggerDettaching
             }
         }
 
+        public T TryGetComponent<T>() where T : class
+        {
+            T component = components.Values.FirstOrDefault(x => x is T) as T;
+            if (component != null)
+                return component;
+
+            component = sharedComponents.Values.FirstOrDefault(x => x is T) as T;
+            if (component != null)
+                return component;
+
+            return null;
+        }
```

Comment:
I see that `TryGetSharedComponent()` now uses this new method.
Besides the fact that LinQ may affect performance if called too frequently (not sure how frequent are calls to `TryGetSharedComponent()`), if this new `TryGetComponent()` method is normally used more for shared components instead of normal components, can we please check the shared components collection first instead of normal components? to avoid that extra LinQ check on most cases.

Reply:
`TryGetSharedComponent()` still have his current implementation so this shouldn't affect it, This method is used for components that implement interfaces. Right now it is called frequently, however, I will add some notes and information to make its use more clear!